src/ReinforcementLearningCore/src/policies/random_policy.jl

```diff
@@ -95,3 +95,5 @@ function RLBase.prob(
         0.0
     end
 end
+
+RLBase.push!(::RandomPolicy, ::Any) = nothing
```

Comment:
MultiAgentPolicy incldues the following 

```
function Base.push!(multiagent::MultiAgentPolicy, ::PreActStage, env::E) where {E<:AbstractEnv}
    for player in players(env)
        push!(multiagent[player], state(env, player))
    end
end
```
In the random policy push!(::RandomPolicy, ::Any) is not defined. So to define the default behavior, this should be nothing as there will be an Error otherwise.
E.g. for TicTacToe Environment:

```
using ReinforcementLearningCore
using ReinforcementLearningBase
using ReinforcementLearningEnvironments
using StableRNGs
using Test


rng = StableRNGs.StableRNG(123)
e = env = TicTacToeEnv()
m = MultiAgentPolicy(NamedTuple((player => RandomPolicy(;rng=rng) for player in players(e))))
hooks = MultiAgentHook(NamedTuple((p => EmptyHook() for p ∈ players(e))))
x = run(m, e, StopAfterEpisode(10), hooks)
==> 
ERROR: LoadError: MethodError: no method matching push!(::RandomPolicy{Nothing, StableRNGs.LehmerRNG}, ::Int64)
```

Reply:
Moreover, push!(::AbstractPolicy, ::Any) should be defined as nothing in the first place to prevent undefined methods to be present for push! as it was added to the api of the environments. Should 
```
RLBase.push!(::AbstractPolicy, ::Any) = nothing
```
 added to `src/ReinforcementLearningBase/src/interface.jl` ?

Reply:
@jeremiahpslewis does this answer your question?

Reply:
yep, then we should add it to `interface.jl`

Reply:
wait, no to the policy part of RLCore

Reply:
on further thought...think `interface.jl` is right place

Reply:
Well, I figured out, that this issue is related to multi_agent.jl.
We only need to call push! to the agents cache if the players separate policies are actually of Subtype Agent. I fixed this in the last commit.
(One can just use a Random Policy or an online policy without any Trace)